### ISSUE TITLE
download.GFDL rewrites "GFDL" anywhere in output path

### ISSUE THREAD

**Issue by infotroph** [PecanProject/pecan#1924]
## Description

```
met_path <- "~/demos/NARRvsGFDL_20180503/data/GFDL"
dir.create(met_path, recursive=TRUE)
first_test <- PEcAn.data.atmosphere::download.GFDL(
	outfolder = met_path,
	start_date = "2007-01-01",
	end_date = "2007-12-31",
	site_id = 753,
	lat.in = 40,
	lon.in = -88)
first_test$file
```

Expected: `~/demos/NARRvsGFDL_20180503/data/GFDL/GFDL.CM3.rcp45.r1i1p1.2007.nc`

Actual: `"~/demos/NARRvsGFDL_CM3_rcp45_r1i1p1_20180503/data/GFDL_CM3_rcp45_r1i1p1/GFDL.CM3.rcp45.r1i1p1.2007.nc"`

## Possible Implementation

To fix, change the offending [gsub call](https://github.com/PecanProject/pecan/blob/develop/modules/data.atmosphere/R/download.GFDL.R#L30) to something that either:

1. creates a new subfolder inside the path that was given, not next to it. I'd strongly prefer this, on the grounds that the function ought not to rewrite user-provided paths *at all*, but I realize "last path component is a prefix rather than an existing directory" is common across the PEcAn met functions. 

2. If it is important to rewrite the path, edit only the the last component, and only by pasting onto the end. Changing components further upstream is unacceptable. Additionally, all functions that rewrite their last component should document this prominently.

**Comment by infotroph** [comment]
Just realized this is exactly the underlying issue that caused #1911, and was fixed in #1914. Can be closed once release/1.5.3 is merged back into develop.

**Comment by infotroph** [comment]
This is closely related to the issue that caused #1911, but the fix for that in #1914 doesn't correct the mangling of directory names higher in the path.
